source/code/scxsystemlib/disk/diskdepend.cpp

```diff
@@ -375,9 +375,21 @@ namespace SCXSystemLib
         SCXCoreLib::SCXHandle<std::wfstream> fsDiskStats(SCXCoreLib::SCXFile::OpenWFstream(LocateProcDiskStats(), std::ios::in));
         fsDiskStats.SetOwner();
         std::wstring line;
-        while ( ! fsDiskStats->eof() && fsDiskStats->is_open() )
+        int counter = 0;
+        while ( fsDiskStats->good() && fsDiskStats->is_open() )
```

Comment:
do we need to add log if it is failbit or badbit when not go inner while?

Reply:
Behaviour of failbit and badbit is not that very well defined like under what condition what exactly will get set across the various OS. For example in my local system, failbit also gets set under eof condition.

Reply:
ok